crates/project/src/lsp_command.rs

```diff
@@ -1552,10 +1557,22 @@ impl LspCommand for GetCompletions {
 
                     let language_registry = language_registry.clone();
                     let language = language.clone();
+
                     LineEnding::normalize(&mut new_text);
                     Some(async move {
                         let mut label = None;
                         if let Some(language) = language.as_ref() {
+                            let max_completion_len =
+                                10.max((-2.8333 * buffer_font_size as f32 + 87.0) as i32) as usize;
+                            if lsp_completion.label.len() > max_completion_len {
+                                lsp_completion.label.truncate(max_completion_len - 3);
+                                lsp_completion.label.push_str("...");
+                            }
+
+                            // Python justifies its LSP text strangely so this workaround is required. I'm not a huge fan either.
+                            if language.name().to_string().as_str() == "Python" {
+                                lsp_completion.label.push_str(" ".repeat(30).as_str());
+                            }
```

Comment:
I don't think this is the right layer to be performing this truncation. We should only truncate in the UI layer.